application/common/components/passwordStore/Google.php

```diff
@@ -178,16 +170,40 @@ public function getMeta($employeeId): UserPasswordMeta
      * Employee ID.
      *
      * @param string $employeeId The Employee ID of the desired user.
-     * @return Google_Service_Directory_User The user record from Google.
+     * @return Google_Service_Directory_User|null The user record from Google.
+     * @throws UserNotFoundException if email not defined or user not found in Google
      */
-    protected function getUser($employeeId)
+    protected function getUser(string $employeeId): ?Google_Service_Directory_User
     {
-        if ($this->findByExternalId === true) {
+        $email = $this->getEmailFromLocalStore($employeeId);
+        $user = $this->getUserByEmail($email);
```

Comment:
It looks like `getUserByEmail()` can't return `null`... is that right? It seems to throw a UserNotFoundException.

In that case, I think we don't need the following if-block. What impact would that have on the intended behavior of the system?

Reply:
Good observation. Looks like my brain is still in Go mode...not catching exceptions.

Reply:
Addressed this in 5.5.1